File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(int a, int b) 
{
    return a>b;
}

int solution(vector<int> citations) {
    int answer = 0;
    
    sort(citations.begin(), citations.end(), compare);
    
    int len = citations.size();
    
    for(int i=0; i < len ; i++)
    {
        if(citations[i] > i) answer +=1;
        else break;
    }
    
    return answer;
}

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    var answer = 0;
    
    citations.sort((a,b) => a-b);
    
    const len = citations.length;
    for(const idx in citations) {
        // 남은 개수 구하고
        const item = citations[idx];
        const cnt = len-idx;
        
        // 현재 위치 값이 개수보다 크거나 같으면 기록
        if(item >= cnt) {
            if(cnt > answer) answer = cnt;
        }
    }
    
    return answer;
}